controllers/rollingupgrade_controller.go

```diff
@@ -602,6 +622,13 @@ func (r *RollingUpgradeReconciler) Reconcile(req ctrl.Request) (ctrl.Result, err
 			logr.Info("Sync map with invalid entry for ", "name", ruObj.Name)
 		}
 	} else {
+
+		if r.maxParallel != nil {
+			logr.Info("trying to acquire semaphore lock...")
+			r.AcquireSemaphore()
```

Comment:
Won't this block the main thread in the running controller when there are more than maxParallel  CRs?

Reply:
Good point, I moved the semaphore lock inside the goroutine, now it should no longer block the main reconcile thread, but the actual `Process()` goroutine instead.

Reply:
BTW, `Reconcile()` on itself is called async, so even if the blocking channel is inside `Reconcile()` it won't block new reconciles from happening (or atleast getting queued)/